modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationEntryLocalServiceTest.java

```diff
@@ -134,6 +142,280 @@ public void testAddCPConfigurationEntry() throws Exception {
+			displayDateHour += 12;
+		}
+
+		CPConfigurationList cpConfigurationList1 =
+			_cpConfigurationListLocalService.addCPConfigurationList(
+				null, _cpConfigurationList.getGroupId(), _user.getUserId(),
+				_cpConfigurationList.getCPConfigurationListId(), false,
+				RandomTestUtil.randomString(), 1, calendar.get(Calendar.MONTH),
+				calendar.get(Calendar.DAY_OF_MONTH),
+				calendar.get(Calendar.YEAR), displayDateHour,
+				calendar.get(Calendar.MINUTE), 0, 0, 0, 0, 0, true);
+
+		Assert.assertTrue(
+			ListUtil.isEmpty(
+				_cpConfigurationEntryLocalService.getCPConfigurationEntries(
+					cpConfigurationList1.getCPConfigurationListId())));
+
+		CPConfigurationEntrySetting cpConfigurationEntrySetting =
+			_cpConfigurationEntrySettingLocalService.
+				fetchCPConfigurationEntrySetting(
+					cpConfigurationEntry1.getCPConfigurationEntryId(),
+					CPConfigurationEntrySettingConstants.TYPE_INDEX_IDS);
+
+		Assert.assertNotNull(cpConfigurationEntrySetting);
+
+		Assert.assertTrue(
+			StringUtil.contains(
+				cpConfigurationEntrySetting.getSetting(),
+				String.valueOf(
+					cpConfigurationList1.getCPConfigurationListId())));
+
+		CPConfigurationList cpConfigurationList2 =
+			_cpConfigurationListLocalService.addCPConfigurationList(
+				null, _cpConfigurationList.getGroupId(), _user.getUserId(),
+				cpConfigurationList1.getCPConfigurationListId(), false,
+				RandomTestUtil.randomString(), 1, calendar.get(Calendar.MONTH),
+				calendar.get(Calendar.DAY_OF_MONTH),
+				calendar.get(Calendar.YEAR), displayDateHour,
+				calendar.get(Calendar.MINUTE), 0, 0, 0, 0, 0, true);
+
+		Assert.assertTrue(
+			ListUtil.isEmpty(
+				_cpConfigurationEntryLocalService.getCPConfigurationEntries(
+					cpConfigurationList2.getCPConfigurationListId())));
+
+		cpConfigurationEntrySetting =
+			_cpConfigurationEntrySettingLocalService.
+				fetchCPConfigurationEntrySetting(
+					cpConfigurationEntry1.getCPConfigurationEntryId(),
+					CPConfigurationEntrySettingConstants.TYPE_INDEX_IDS);
+
+		Assert.assertNotNull(cpConfigurationEntrySetting);
+
+		Assert.assertTrue(
+			StringUtil.contains(
+				cpConfigurationEntrySetting.getSetting(),
+				String.valueOf(
+					cpConfigurationList2.getCPConfigurationListId())));
+
+		externalReferenceCode = RandomTestUtil.randomString();
+
+		CPConfigurationEntry cpConfigurationEntry2 =
+			_cpConfigurationEntryLocalService.addCPConfigurationEntry(
+				externalReferenceCode, _user.getUserId(),
+				cpConfigurationList2.getGroupId(),
+				_portal.getClassNameId(CPDefinition.class),
+				_cpDefinition.getCPDefinitionId(),
+				cpConfigurationList2.getCPConfigurationListId(), 0, "123", true,
+				0, "cpde", 1.0, true, true, true, 1.0, "lowstoc",
+				BigDecimal.TEN, BigDecimal.ONE, BigDecimal.ONE, BigDecimal.ONE,
+				true, true, 1.0, true, true, true, 1.0, 1.0);
+
+		Assert.assertEquals(
+			externalReferenceCode,
+			cpConfigurationEntry2.getExternalReferenceCode());
+
```

Comment:
same consideration here... in case we remove this checks we can inline the random instead of having the variable